tools/lsp/common/rename_component.rs

```diff
@@ -1947,4 +2397,693 @@ export { Foo as User4Fxx }
             }
         }
     }
+
+    #[track_caller]
+    fn node_eq(id1: &SyntaxNode, id2: &SyntaxNode) {
+        assert_eq!(id1.kind(), id2.kind());
+        assert_eq!(id1.source_file.path(), id2.source_file.path());
+        assert_eq!(id1.text_range(), id2.text_range());
+    }
+
+    #[track_caller]
+    fn find_declaration_node_by_comment(
+        document_cache: &common::DocumentCache,
+        document_path: &Path,
+        suffix: &str,
+    ) -> DeclarationNode {
+        let name = find_node_by_comment(document_cache, document_path, suffix);
+        find_declaration_node(document_cache, name).unwrap()
+    }
+
+    #[track_caller]
+    fn find_identifier_by_comment(
+        document_cache: &common::DocumentCache,
+        document_path: &Path,
+        suffix: &str,
+    ) -> syntax_nodes::DeclaredIdentifier {
+        find_declaration_node_by_comment(document_cache, document_path, suffix)
+            .as_declared_identifier()
+    }
+
+    #[test]
+    fn test_rename_component_from_use() {
+        let document_cache = test::compile_test_with_sources(
+            "fluent",
+            HashMap::from([(
+                Url::from_file_path(test::main_test_file_name()).unwrap(),
+                r#"
+component Foo /* <- TEST_ME_TARGET1 */ { @children }
+
+export { Foo /* <- TEST_ME_1 */}
+
+enum Xyz { Foo, Bar }
+
+struct Abc { Foo: Xyz }
+
+component Baz {
+    Foo /* <- TEST_ME_2 */ { }
+}
+
+component Foo /* <- TEST_ME_TARGET2 */ inherits Foo /* <- TEST_ME_3 */ {
+    Foo /* <- TEST_ME_4 */ { }
+}
+
+export component Bar inherits Foo /* <- TEST_ME_5 */ {
+    Foo /* <- TEST_ME_6 */ { }
+    Rectangle {
+        Foo /* <- TEST_ME_7 */ { }
+        Foo := Baz { }
+    }
+
+    if true: Rectangle {
+        Foo /* <- TEST_ME_8 */ { }
+    }
+
+    if false: Rectangle {
+        Foo /* <- TEST_ME_9 */ { }
+    }
+
+    function Foo(Foo: int) { Foo + 1; }
```

Comment:
The test should also test that none of the `Foo` here matches with a component.

Reply:
None of this should trigger for function names yet, but true, that should be tested.